src/bin/cargo-miri.rs

```diff
@@ -519,25 +556,27 @@ fn inside_cargo_rustc() {
     let sysroot = std::env::var("MIRI_SYSROOT").expect("The wrapper should have set MIRI_SYSROOT");
 
     let rustc_args = std::env::args().skip(2); // skip `cargo rustc`
-    let mut args: Vec<String> =
-        rustc_args.chain(Some("--sysroot".to_owned())).chain(Some(sysroot)).collect();
-    args.splice(0..0, miri::miri_default_args().iter().map(ToString::to_string));
+
+    let in_build_script = is_build_dep(std::env::args().skip(2));
+
+    let mut args = if in_build_script {
+        rustc_args.collect()
```

Comment:
Not setting `--sysroot` here is crucially important, right, because this generates actual code? Please add a comment explaining that.

Reply:
Added